### PR TITLE
MSI: 426: Update StockItemDataChecker

### PR DESCRIPTION
**Pull request by RomaKis**
### Fixed Issues (if relevant)
1. magento-engcom/msi#426: Update StockItemDataChecker

### CODE REVIEW

File: dev/tests/integration/testsuite/Magento/CatalogInventory/Model/StockItemSave/StockItemDataChecker.php [hunk 1]
```diff
@@ -166,21 +178,24 @@ private function assertArrayContains(array $expected, array $actual)
...
-        $sourceItem = reset($sourceItems);
-        Assert::assertEquals(
-            $expectedData[StockItemInterface::QTY],
-            $sourceItem->getQuantity()
-        );
-        Assert::assertEquals(
-            $expectedData[StockItemInterface::IS_IN_STOCK],
-            (int)$sourceItem->getStatus()
-        );
+        // soft dependency in tests because we don't have possibility replace tests from different modules
```
Comment (naydav) [inline dev/tests/integration/testsuite/Magento/CatalogInventory/Model/StockItemSave/StockItemDataChecker.php]: Need to move this check inti `checkStockItemData` and wrap `$this->checkIntegrityWithInventory($product, $expectedData);` call

Comment (RomaKis) [reply dev/tests/integration/testsuite/Magento/CatalogInventory/Model/StockItemSave/StockItemDataChecker.php]: Done.

**Review by orlangur — APPROVED** [review]

File: dev/tests/integration/testsuite/Magento/CatalogInventory/Model/StockItemSave/StockItemDataChecker.php [hunk 2]
```diff
@@ -103,7 +115,10 @@ public function checkStockItemData($sku, array $expectedData)
         $productLoadedByModel = $this->productFactory->create();
         $productLoadedByModel->load($product->getId());
         $this->doCheckStockItemData($product, $expectedData);
-        $this->checkIntegrityWithInventory($product, $expectedData);
+        // soft dependency in tests because we don't have possibility replace tests from different modules
+        if ($this->moduleManager->isEnabled('Magento_Inventory')) {
```
Comment (naydav) [inline dev/tests/integration/testsuite/Magento/CatalogInventory/Model/StockItemSave/StockItemDataChecker.php]: According to that MSI will be stored in separate repository
Need to revert all of this changes and copy all of tests to MSI current repository

**Review by naydav — CHANGES_REQUESTED** [review]

**Comment by RomaKis** [comment]
Close due to low priority.

**Pull request closed without merging**